salt/states/zabbix_host.py

```diff
@@ -336,21 +351,33 @@ def _interface_format(interfaces_data):
         elif not hostinterfaces and interfaces:
             update_interfaces = True
 
-        cur_inventory = __salt__["zabbix.host_inventory_get"](
-            hostids=hostid, **connection_args
-        )
-        if cur_inventory:
-            # Remove blank inventory items
-            cur_inventory = {k: v for k, v in cur_inventory.items() if v}
-            # Remove persistent inventory keys for comparison
-            cur_inventory.pop("hostid", None)
-            cur_inventory.pop("inventory_mode", None)
-
-        if not cur_inventory:
-            if new_inventory:
-                update_inventory = True
-        elif set(cur_inventory) != set(new_inventory):
-            update_inventory = True
+        # if inventory param is empty leave inventory as is don't compare it
+        # if inventory_mode is '-1', the inventory will be erased, why compare it?
+        if inventory is not None and inventory_mode != "-1":
+            cur_inventory = __salt__["zabbix.host_inventory_get"](
+                hostids=hostid, **connection_args
+            )
+            if cur_inventory:
+                cur_inventory.pop("hostid", None)
+                cur_inventory.pop("inventory_mode", None)
+                if new_inventory:
+                    for key in new_inventory:
+                        if (
+                            key in cur_inventory
+                            and new_inventory[key] != cur_inventory[key]
+                        ):
+                            update_inventory = True
+                            break
+                else:
+                    # remove the empty keys in current inventory
+                    cur_inventory = {k: v for k, v in cur_inventory.items() if v}
+                    # if current inventory any no empty key and new inventory is empty
+                    if not cur_inventory:
+                        update_inventory = True
```

Comment:
**suggestion**: check out [`salt.utils.dictdiffer`](https://github.com/saltstack/salt/blob/master/salt/utils/dictdiffer.py), it's built for this purpose

Reply:
I'll check it.

Reply:
Nice, the code was changed to use `salt.utils.dictdiffer.diff()`